### ISSUE TITLE
This is awesome. What's its license?

### ISSUE THREAD

**Issue by i336** [openzfsonwindows/ZFSin#6]
This just got posted to Hacker News (https://news.ycombinator.com/item?id=15549190), where a ton of people are going to learn about it. It'll probably end up on Reddit soon too, if it hasn't already.

I don't see a license file anywhere. Now would be a good time to clarify the license.

AFAIK, the difference between BSD and MIT is that (IIRC!!!) BSD requires you to cite the original author(s), while MIT does not.

When mentioning licenses I usually link to the https://en.wikipedia.org/wiki/Sybase_Open_Watcom_Public_License as well, primarily because it makes for good food for thought. (In certain ways it's stricter than the GPL, but these differences also make it GPL incompatible).

**Comment by bitofhope** [comment]
> AFAIK, the difference between BSD and MIT is that (IIRC!!!) BSD requires you to cite the original author(s), while MIT does not.

MIT absolutely does.

```
The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.
```

This seems to include CDDL licensed ZFS code, which must remain CDDL. Not sure what the @lundman wants to put their work under, though.

**Comment by lundman** [comment]
Most sources are CDDL, as they are from OpenZFS of course. I personally have no reason to change it. :)

**Comment by i336** [comment]
**@bitofhope**: Thanks so much for the clarification. I read somewhere that one has an advertising clause and one doesn't. It looks like neither do, and I have some reading/digging to do to figure out what I misread (or wrongly believed).

**@lundman:** Somehow I managed to get it into my head that this was a from-scratch driver, haha. Don't think I thought that one through :) D'oh, of course it's going to already bring a license along. Thanks for the clarification.

**Comment by bitofhope** [comment]
**tl;dr:** sorry for derailing the issue

@i336 The thing about the BSD license is that it's actually not just one license but three or more different ones. The ones usually counted are the 4-clause, 3-clause, and 2-clause BSD Licenses.

The 4-clause one has the advertising clause. If I advertise my software having a feature provided by a 4cBSD piece of code, I have to credit the copyright holder in the same ad. The 4-clause is not very popular these days.

The 3-clause one drops the advertising clause, but still keeps the endorsement clause. You can't use the original author's name in promoting your software without excplicit permission.

The 2-clause BSD is functionally near-identical with the MIT license, except the MIT license explicitly allows sublicensing while it's debated with 2cBSD.

CDDL is a different, more complicated beast altogether.

**Comment by fire** [comment]
I don't think it's possible to change this project from cddl.

Could you archive this issue since most of the work is under cddl? Perhaps the overlay app can be under a different license, but it'll still have to link the zfs library or an api.

**Comment by TheCakeIsNaOH** [comment]
@lundman Can this be closed? 

There is a license file in the repository, and the installer includes a license page.
